skimage/measure/entropy.py

```diff
@@ -0,0 +1,39 @@
+from scipy.stats import entropy as scipy_entropy
+from .._shared.utils import assert_nD
+
+
+def shannon_entropy(image, base=2):
+    """
+    Calculate the Shannon entropy of an image.
+
+    The Shannon entropy is defined as S = -sum(pk * log(pk)),
+    where pk are the number of pixels of value k.
```

Comment:
With this definition, it's not obvious to me how the entropy is computed on an image of floats. However, I just checked that the entropy is the same whether the image is made of ints or floats.